### PR TITLE
refactor(icon): add icon component with props support

### PR DESCRIPTION
**Pull request by patrickliu-sa**
As the old unified icon component has room to be enhanced. I've replicated that to a new icon component with prop support:

icon component

### BREAKING CHANGE:

Take the jobAd icon as example,

#### Before:
```js
<Icons iconName="jobAd" />
```
#### After:
```js
<Icon type="jobAd" />
```

#### Before:

```js
import { 
  Icons
} from 'seekasia-style-guide/react';
```

#### After:

```js
import { 
  Icon
} from 'seekasia-style-guide/react';
```

#### - Small (16px x 16px):
```js
<Icon type="jobAd" size="small" /> 
```

#### - Large (36px x 36px):
```js
<Icon type="jobAd" size="large"/> 
```

#### - 90 degree rotated: 
```js
<Icon type="jobAd" rotation="90deg" />
```

#### - 180 degree rotated: 
```js
<Icon type="jobAd" rotation="180deg" />
```

#### - 90 degree rotated (counter-clockwise): 
```js
<Icon type="jobAd" rotation="-90deg" />
```

#### - 180 degree rotated (counter-clockwise): 
```js
<Icon type="jobAd" rotation="-180deg" />
```

#### - Smooth rotation 
```js
<Icon smoothRotate type="jobAd" />
```
(which can be use in conjunction with `rotation` prop)

#### - Animation
```js
<Icon animation="bounce" type="jobAd" />
<Icon animation="popIn" type="jobAd" />
```

### CODE REVIEW

**Comment by patrickliu-sa** [comment]
Preview of this new icon component:
http://seek-asia-style-guide--a5c63278eb6ec2170c529c45dfd98f30714c867f.surge.sh/icon

File: react/Icon/Icon.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+export default function Icon({
+  type,
+  className,
+  small,
+  large,
+  title,
+  rotate90,
+  rotate180,
+  rotate90cc,
+  rotate180cc,
```
[the remark this answers is not in the document]
Comment (mengtzu) [reply react/Icon/Icon.js]: Sorry, I didn't understand that it was in conjunction with the `animate` prop 👍

**Comment by patrickliu-sa** [comment]
I can see BRAID use more inline css-in-js than SEEK Asia style guide (which is not a very bad thing after all). What I'm try to do here is encapsulate animation as flexible preset for the ease of adaption.  Inline keyframes management is still not easy to do...

File: react/Icon/Icon.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrickliu-sa) [reply react/Icon/Icon.js]: And now they've been grouped as `rotation: PropTypes.oneOf(['', '90deg', '-90deg', '180deg', '-180deg'])`

**Comment by patrickliu-sa** [comment]
New preview snapshot:
https://seek-asia-style-guide--56f7512ce2ff0df7f02bd9518302d70ef98f01a1.surge.sh/icon/

**Comment by mengtzu** [comment]
Looks good, but does this deprecate the `Icons` page in the demo site?  Might be good to remove it.

**Comment by patrickliu-sa** [comment]
I hope to do that now, but found there are some `Icons` are not covered in the style guide set.  Better green light this PR first and take another PR for tidy up

**Pull request merged**